src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -70,7 +70,7 @@ public async Task<string> GetCurrentUserTimeZoneIdAsync()
         var key = GetCacheKey(userName);
         var timeZoneId = await _distributedCache.GetStringAsync(key);
 
-        if (string.IsNullOrEmpty(timeZoneId))
+        if (timeZoneId is null)
```

Comment:
IsNullOrEmpty seems better since the serialization might use "" in the store. I am not sure Redis can store "null"

Reply:
That's what I found and trying to fix.

Reply:
I am doing local changes, let me add it

Reply:
Ok, here we don't want to go and setStringAsync only when a cache entry is not found else it will pass on every request which defeats the purpose. So this is why I'm using `timeZoneId is null` which I think is the perf fix. Then, I'm just going to cache it as a String.Empty() which should also be fine since it finds the cache entry by Key.

Reply:
Oh well I pushed my changes but I will let you work on the remaining then :smile:

Reply:
Magic value could be "null" too. But seems like it works because we do a String.IsNullOrEmpty on the GetUserTimeZoneAsync method